vochain/scrutinizer/scrutinizer.go

```diff
@@ -97,26 +99,40 @@ func NewScrutinizer(dbPath string, app *vochain.BaseApplication) (*Scrutinizer,
 		return nil, err
 	}
 	startTime := time.Now()
-	envelopes, err := s.db.Count(&indexertypes.VoteReference{}, &badgerhold.Query{})
-	if err != nil {
-		return nil, fmt.Errorf("could not count the total envelopes: %w", err)
+
+	txCountStore := new(indexertypes.CountStore)
+	if err = s.db.FindOne(txCountStore,
+		badgerhold.Where(badgerhold.Key).Eq(indexertypes.Transactions)); err != nil {
+		log.Warnf("could not get the transaction count: %v", err)
 	}
-	entities, err := s.db.Count(&indexertypes.Entity{}, &badgerhold.Query{})
-	if err != nil {
-		return nil, fmt.Errorf("could not count the total entities: %w", err)
+	envelopeCountStore := new(indexertypes.CountStore)
+	if err = s.db.FindOne(envelopeCountStore,
+		badgerhold.Where(badgerhold.Key).Eq(indexertypes.Envelopes)); err != nil {
+		log.Warnf("could not get the envelope count: %v", err)
 	}
-	processes, err := s.db.Count(&indexertypes.Process{}, &badgerhold.Query{})
-	if err != nil {
-		return nil, fmt.Errorf("could not count the total processes: %w", err)
+	processCountStore := new(indexertypes.CountStore)
+	if err = s.db.FindOne(processCountStore,
+		badgerhold.Where(badgerhold.Key).Eq(indexertypes.Processes)); err != nil {
+		log.Warnf("could not get the process count: %v", err)
 	}
+	entityCountStore := new(indexertypes.CountStore)
+	if err = s.db.FindOne(entityCountStore,
+		badgerhold.Where(badgerhold.Key).Eq(indexertypes.Entities)); err != nil {
+		log.Warnf("could not get the entity count: %v", err)
+	}
+
+	s.countTotalTransactions = txCountStore.Count
```

Comment:
Since now we have the counters stored on the persistent KV, I don't think its worth to keep these in-memory caches.

Reply:
Are you proposing to have a db transaction every time we want to access or update these values, i.e. every block?